### PR TITLE
Add select() alternative to debugger socket transport example

### PR DESCRIPTION
**Pull request by nkolban**

### CODE REVIEW

**Comment by svaarala** [comment]
@nkolban Some codepolicy issues fail:

```
vim +27    examples/debug-trans-socket/duk_trans_socket_unix.c : #ifndef found, #if !defined is preferred - #ifndef USE_SELECT
vim +272   examples/debug-trans-socket/duk_trans_socket_unix.c : #ifdef found, #if defined is preferred - #ifdef USE_SELECT
vim +289   examples/debug-trans-socket/duk_trans_socket_unix.c : #ifdef found, #if defined is preferred - #ifdef USE_SELECT
vim +14    examples/debug-trans-socket/duk_trans_socket_unix.c : trailing whitespace - *
vim +16    examples/debug-trans-socket/duk_trans_socket_unix.c : trailing whitespace - *
vim +20    examples/debug-trans-socket/duk_trans_socket_unix.c : trailing whitespace - *
*** Total: 6 problems
```

**Comment by nkolban** [comment]
Thank you sir ... sorry about that and GLAD that we have a policy.  All good ... will fix up.

**Comment by svaarala** [comment]
Thanks -- please squash commits, and also add yourself to AUTHORS.rst :)

**Comment by nkolban** [comment]
Sorry to be a dummy ... but can you give me a recipe or pointer to "squashing commits"?

**Comment by svaarala** [comment]
There are many ways to do it (see e.g. https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase). I usually use:
- git checkout my-feature-branch
- git rebase -i master
- git push --force-with-lease

When the commit list is open, it lists all commits with "pick" by default. You can reorder the lines if necessary. To squash a commit to the one above, change "pick" to "squash". When you save and exit from the commit list, git will replay the commits. For squashed commits you are requested to combine the commit messages into one.

File: examples/debug-trans-socket/duk_trans_socket_unix.c [hunk 1]
```diff
@@ -11,20 +11,20 @@
...
- * 
+ *
  */
 #include <stdio.h>
 #include <string.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <unistd.h>
-#ifndef USE_SELECT
+#if !defined USE_SELECT
```
Comment (svaarala) [inline examples/debug-trans-socket/duk_trans_socket_unix.c]: Please use parens here, e.g. `#if !defined(USE_SELECT)`. The current convention is described here: https://github.com/svaarala/duktape/blob/master/doc/code-issues.rst#preprocessor-ifdef-vs-if-defined.

File: examples/debug-trans-socket/duk_trans_socket_unix.c [hunk 2]
```diff
@@ -11,20 +11,20 @@
...
  * file.  For example:
- * 
+ *
  * #define USE_SELECT
- * 
+ *
  * If set, instead of leveraging "poll()", this code will leverage
  * "select()".   Discussions on "poll()" vs "select()" can be read
  * about here: https://daniel.haxx.se/docs/poll-vs-select.html
- *
```
Comment (svaarala) [inline examples/debug-trans-socket/duk_trans_socket_unix.c]: Really small nitpicking but it'd be nice if you could rework to match https://github.com/svaarala/duktape/blob/master/doc/code-issues.rst#comment-styles.

**Comment by svaarala** [comment]
Hmm, there are two commits and one of them has a date as a commit message so the rebase wasn't successful.

Re: commit messages, using rouhgly http://chris.beams.io/posts/git-commit/ would be nice. Just giving the GH issue number ("implement #nnn") doesn't say much when you browse commit messages.

**Comment by nkolban** [comment]
Yeah ... I'm making a hash of this.  I don't use Git very much.  What I think we have is the following:

This I forked the master duktape repository.  I then cloned to my PC, made some changes and issued a commit of those changes using "#<Issue number>" as the commit message.  This should link the the commit against the issue that the commit resolves.

We then found that the changes I made weren't in policy.  So I made the changes to the source to bring them into policy.  My thinking was that I would then "squash" those changes together.  So after making the changes to the source, I again issued a commit and a push ... and THEN tried to run the recipe you supplied to squash them together ... however it seems that by this time, the changes had already been pushed through the Pull Request and ended up on your server.  I don't know where we are now :-(

**Comment by svaarala** [comment]
You can just `git rebase -i master` again and then check (e.g. with git log, I use the `tig` command) that the commits came out right. Then `git push --force-with-lease` which will forcibly update the commits in the pull.

**Comment by svaarala** [comment]
Closed as unmerged, @nkolban will issue a new pull.

**Pull request closed without merging**